activedata_etl/transforms/jscov_to_es.py

```diff
@@ -108,55 +164,52 @@ def process_source_file(parent_etl, count, obj, task_cluster_record):
         orphan_covered = orphan_covered - method_covered
         orphan_uncovered = orphan_uncovered - method_uncovered
 
-        new_record = set_default(
-            {
-                "test": {
-                    "name": test_name,
-                    "url": obj.testUrl
-                },
-                "source": {
-                    "language": "js",
-                    "file": file_info,
-                    "method": {
-                        "name": method_name,
-                        "covered": sorted(method_covered),
-                        "uncovered": sorted(method_uncovered),
-                        "total_covered": len(method_covered),
-                        "total_uncovered": len(method_uncovered),
-                        "percentage_covered": method_percentage_covered,
-                    }
-                },
-                "etl": {
-                    "id": count(),
-                    "source": parent_etl,
-                    "type": "join",
-                    "machine": machine_metadata,
-                    "timestamp": Date.now()
-                }
-            },
-            task_cluster_record
-        )
-        key = etl2key(new_record.etl)
-        yield {"id": key, "value": new_record}
+        # Record method coverage info
+        record["source"]["method"] = {
+            "name": method_name,
+            "covered": sorted(method_covered),
+            "uncovered": sorted(method_uncovered),
+            "total_covered": len(method_covered),
+            "total_uncovered": len(method_uncovered),
+            "percentage_covered": method_percentage_covered,
+        }
+
+        # Timestamp this record
+        record["etl"]["timestamp"] = Date.now()
+
+        key = etl2key(record.etl)
+        yield {"id": key, "value": record}
 
     # a record for all the lines that are not in any method
     # every file gets one because we can use it as canonical representative
+    # Record method coverage info
+    record["source"]["method"] = {
+        "covered": sorted(orphan_covered),
+        "uncovered": sorted(orphan_uncovered),
+        "total_covered": len(orphan_covered),
+        "total_uncovered": len(orphan_uncovered),
+        "percentage_covered": len(orphan_covered) / max(1, (len(orphan_covered) + len(orphan_uncovered))),
+    }
+
+    # Timestamp this record
+    record["etl"]["timestamp"] = Date.now()
+
+    key = etl2key(record.etl)
+    yield {"id": key, "value": record}
+
+
+def create_record(parent_etl, count, filename, covered, uncovered, task_cluster_record):
```

Comment:
I am skeptical that this function is required.  On one hand, it does reduce redundancy, but on the other hand, it is not obvious what the total coverage record is.  

The first step is to simplify this function: put it in the lexical scope of `process_jscov_artifact`; then you will not require the `task_cluster_record` parameter.

Reply:
Yeah I agree. Especially when using it in process_source_file, it's hard to see what the coverage record would look like. If I were to put create_record in process_jscov_artifact, process_source_file function won't be using create_record to create record, correct?